src/main/java/bio/terra/service/snapshotbuilder/SnapshotAccessRequestModel.java

```diff
@@ -33,18 +26,35 @@ public record SnapshotAccessRequestModel(
   @VisibleForTesting
   static String generateSummaryForCriteria(
       SnapshotBuilderProgramDataListCriteria criteria, SnapshotBuilderSettings settings) {
+    SnapshotBuilderProgramDataListOption listOption =
+        (SnapshotBuilderProgramDataListOption)
+            settings.getProgramDataOptions().stream()
+                .filter(
+                    programDataOption ->
+                        Objects.equals(programDataOption.getId(), criteria.getId())
+                            && programDataOption.getKind() == SnapshotBuilderOption.KindEnum.LIST)
+                .findFirst()
+                .orElseThrow(
+                    () ->
+                        new InternalServerErrorException(
+                            String.format("No value found for criteria ID %d", criteria.getId())));
```

Comment:
It's a minor thing but I think `formatted()` is nice for these cases
```suggestion
                           "No value found for criteria ID %d".formatted(criteria.getId())));
```